### ISSUE TITLE
Emitter, receiver, subject and proper names used in Solresol to get rhythms

### ISSUE THREAD

**Issue by josepssv** [josepssv/solresol_rhythm#1]
Emitter, receiver, subject and proper names used to achieve rhythm. That is the idea to implement, to advance in the musical level of Solresol.

The subject will always be the sender and the receiver first. In the language of communication "sender/receiver" is what gives meaning and context to any sentence. You should always say "I" and then "you" or "we" although it is always omitted. For example:

Our sentence is: "The city is there." A parsing tells us that "The city" is the subject but in reality the complete sentence would be "I am telling you that the city is there."

Let's take this as a first premise.

Solresol does not have 'developed the proper noun but it does have the pronouns in the first place. Let's use the pronouns to start developing the rhythm.

But it would be very boring in musical terms if all the songs or messages had Dore. To solve this we can use the pronouns first and then the proper nouns as rhythm influencers that affect the melody.

This is the second premise to continue this thread.

**Comment by josepssv** [comment]
I will use this nomenclature naming to separate consonants, vowels, syllables, words and phrases C V - -- .  Although it could be otherwise, it is about how those consonants, vowels, syllables, picked up from a pronoun or a proper name in any language would affect the melody of a Solresol phrase or word.

Now the issue is how to develop that: the pronoun I has only one syllable in almost all languages.  My first choice would be to fit all the notes of a Solresol word into a measure. For example:

If the proper noun taken in the original language has one syllable then all the notes of the word would all fit into one measure.  "I" would make a word like Browse Lamifare only occupy one bar of 4/4, 2/4 or 3/4.  I'll simplify to 4/4 What variety of musical note beats fit into a measure? Using the standardized note beats
"1","2","4","8","16","32","64","dd2","d2","dd4","d4","4t","dd8","d8","8t","16t","dd16","d16"
in total there are 134 variations possible combinations for the measure:

[["2","4","8","8"],["2","4","16","d8"],["2","4","d8","16"],["2","8","4","8"],["2","8","8","4"],["2","8","d8","d8"],["2","16","4","d8"],["2","16","16","d4"],["2","16","d4","16"],["2","16","d8","4"],["2","d4","16","16"],["2","d8","4","16"],["2","d8","8","d8"],["2","d8","16","4"],["2","d8","d8","8"],["4","2","8","8"],["4","2","16","d8"],["4","2","d8","16"],["4","4","4","4"],["4","4","8","d4"],["4","4","d4","8"],["4","8","2","8"],["4","8","4","d4"],["4","8","8","2"],["4","8","d4","4"],["4","16","2","d8"],["4","16","d8","2"],["4","d4","4","8"],["4","d4","8","4"],["4","d4","d8","d8"],["4","d8","2","16"],["4","d8","16","2"],["4","d8","d4","d8"],["4","d8","d8","d4"],["8","2","4","8"],["8","2","8","4"],["8","2","d8","d8"],["8","4","2","8"],["8","4","4","d4"],["8","4","8","2"],["8","4","d4","4"],["8","8","2","4"],["8","8","4","2"],["8","8","d4","d4"],["8","16","16","d2"],["8","16","d2","16"],["8","d2","16","16"],["8","d4","4","4"],["8","d4","8","d4"],["8","d4","d4","8"],["8","d8","2","d8"],["8","d8","d8","2"],["16","2","4","d8"],["16","2","16","d4"],["16","2","d4","16"],["16","2","d8","4"],["16","4","2","d8"],["16","4","d8","2"],["16","8","16","d2"],["16","8","d2","16"],["16","16","2","d4"],["16","16","8","d2"],["16","16","d2","8"],["16","16","d4","2"],["16","d2","8","16"],["16","d2","16","8"],["16","d4","2","16"],["16","d4","16","2"],["16","d4","d4","d8"],["16","d4","d8","d4"],["16","d8","2","4"],["16","d8","4","2"],["16","d8","d4","d4"],["32","32","d2","d8"],["32","32","d8","d2"],["32","d2","32","d8"],["32","d2","d8","32"],["32","d2","8t","8t"],["32","d8","32","d2"],["32","d8","d2","32"],["32","8t","d2","8t"],["32","8t","8t","d2"],["d2","8","16","16"],["d2","16","8","16"],["d2","16","16","8"],["d2","32","32","d8"],["d2","32","d8","32"],["d2","32","8t","8t"],["d2","d8","32","32"],["d2","8t","32","8t"],["d2","8t","8t","32"],["d4","2","16","16"],["d4","4","4","8"],["d4","4","8","4"],["d4","4","d8","d8"],["d4","8","4","4"],["d4","8","8","d4"],["d4","8","d4","8"],["d4","16","2","16"],["d4","16","16","2"],["d4","16","d4","d8"],["d4","16","d8","d4"],["d4","d4","8","8"],["d4","d4","16","d8"],["d4","d4","d8","16"],["d4","d8","4","d8"],["d4","d8","16","d4"],["d4","d8","d4","16"],["d4","d8","d8","4"],["d8","2","4","16"],["d8","2","8","d8"],["d8","2","16","4"],["d8","2","d8","8"],["d8","4","2","16"],["d8","4","16","2"],["d8","4","d4","d8"],["d8","4","d8","d4"],["d8","8","2","d8"],["d8","8","d8","2"],["d8","16","2","4"],["d8","16","4","2"],["d8","16","d4","d4"],["d8","32","32","d2"],["d8","32","d2","32"],["d8","d2","32","32"],["d8","d4","4","d8"],["d8","d4","16","d4"],["d8","d4","d4","16"],["d8","d4","d8","4"],["d8","d8","2","8"],["d8","d8","4","d4"],["d8","d8","8","2"],["d8","d8","d4","4"],["8t","32","d2","8t"],["8t","32","8t","d2"],["8t","d2","32","8t"],["8t","d2","8t","32"],["8t","8t","32","d2"],["8t","8t","d2","32"]] 

This gives an idea of the problem

But the problem would increase enormously if we want to place 5 or 6 notes in a 4/4 time signature then the number of alternatives would be for the case of 5 notes: 775 possibilities some of them very very fast, for example ["32", "32", "d2", "d8"] at very high BPM almost indistinguishable. I'm going to look for a way to hear that!

It would be a matter of placing Lamifare in one of these places of time, but which one to choose?

The solution can be given by the pronoun or proper name in any language.

We are going to use "Me" but what if we use "I" or other?... I am considering this. First reduce to the code CV-

I = V one vowel one syllable 
Me = CV ccnsonant and vowel in one syllable 
Myself = CV-CVCC consonants and vowels in two syllables

From the 4/4 normalized beat matrix the simplest would be ["4", "4", "4", "4"] i.e. four quarter notes. We can take into account that a quarter note can be divided into other parts in turn, for example: How many parts can a quarter note be divided into?
[
  ["4"],
  ["8", "8"],
  ["8", "16", "16"],
  ["16", "8", "16"],
  ["16", "16", "8"],
  ["8", "16", "32", "32"],
  ["8", "32", "16", "32"],
  ["8", "32", "32", "16"],
  ["16", "8", "32", "32"],
  ["16", "32", "8", "32"],
  ["16", "32", "32", "8"],
  ["32", "8", "16", "32"],
  ["32", "8", "32", "16"],
  ["32", "16", "8", "32"],
  ["32", "16", "32", "8"],
  ["32", "32", "8", "16"],
  ["32", "32", "16", "8"],
  ["8", "16", "32", "32"],
  ["8", "32", "16", "32"],
  ["8", "32", "32", "16"],
  ["16", "8", "32", "32"],
  ["16", "32", "8", "32"],
  ["16", "32", "32", "8"],
  ["32", "8", "16", "32"],
  ["32", "8", "32", "16"],
  ["32", "16", "8", "32"],
  ["32", "16", "32", "8"],
  ["32", "32", "8", "16"],
  ["32", "32", "16", "8"],
  ["16", "16", "16", "16"]]

  in other words, at the extreme we would have each quarter note divided into four ["32", "32", "32", ...] ] this would sound very fast, a machine gun, 32 notes per bar would be the maximum and the minimum one note per bar


Now you have to introduce two "Dore" notes in the bar, another difficulty. It could be up to two notes in length 32 which would leave a large gap of silence in the bar

... will continue

**Comment by josepssv** [comment]
Totally confused now

I'm going to clear up those disjointed preconceptions I'm looking for a way out of.
These are perhaps single sentences
First is the rhythm and then the melody, in this I'm almost sure.

But can it be a totally free rhythm and then the Solresol notes are incorporated. It is necessary to take into account that they must be 1, 2, 3, 3, 4 or 5 and they cannot be disordered because they would lose the meaning? They can be removed in several measures or shrink a lot in one. It also happens as in verbal language if we add one after the other, different meanings arise, that is why the recommendation to separate them, but in music and poetry these games and repetitions are interesting for the text to pick up the rhythm. For example: I will dance all night da da da dance all night long. All night long, all night, all night long, all night. 

So we have here another premise for music we do not need a rigorous communication of the message as we do for verbal or written reasoning. I see Solresol now as an atmosphere of color, an ecosystem that as a whole transmits a global message, that envelops, a tonal atmosphere, that will be very different depending on what term we are using. But what about rhythm?

To be a rhythm it must be a repetitive structure, a template. In principle a metronome marks the rhythm but with stronger or softer beats it is more dynamic and that is what it is all about. This dynamism of accents in Solresol defines a part of the grammar being able to express verbs or adjectives of the same concept. Shall we take this into account? Let's take an example:

Solsifasi GBFB Innovation :  
The accent can be given by the pitch of the note in octaves or by the intensity of the stroke: 
 And change the meaning to verb, noun, adjective and adverb.  Innovation, Innovate, Innovated, Innovatedly (these changes of meaning when accentuating in rhythm are curious).
The accent especially is also given by the duration of the note.

4 4 4 4  a 4/4 time signature, for accenting we have many possibilities, (although this may indicate gender or plural in the grammar. I must check that)  2 4 8 8 ,  2 4 16 d8,  2 d8 d8 8, etc


 
- G4 B3 F3 B3 would be like Sólsifasi the rhythm | Sól si fa si | ---> Perfect to carry the beat to the first beat 4/4  -   2 4 8 8  - Verb

- G3 B4 F3 B3 would be like Solsífasi the rhythm | x x x x Sol | sí fa si x |  --Does not fit in a 4/4 time signature but does fit with two rests in a 3/4 - 4 2 8 8 - Noun  

- G3 B3 F4 B3 would be like Solsifási the rhythm | x x x Sol si | fá si x x x x |  --> Perfect for a 2/4 time signature, in this case it would be: Sólsifási or stressing on the last Solsífasí  - 8 4 2 8 -  Adjective

- G3 B3 F3 B4 would be like Solsifasí the rhythm | x Sol si fa | sí x x x x x |  --  8 8 4 2 -  Adverb

...etc..

**Comment by abu-rih** [comment]
Great Work José :)

**Comment by josepssv** [comment]
I have gone a bit off topic but I can use this reflection to develop something more related. If we solve this problem we will have made good progress:

We are going to build a rhythm from a name and then the Solresol will adapt to the rhythm. The first part alone is already complex, and you Abu already have a lot of experience.

To build a rhythm from a text we have seen that the accent is very important. Although I have an idea, any other idea is welcome. But first I have to decide how to incorporate the accent. So far we only had Consonant, Vowel, Syllable, Word.  C V - -- now we are going to add vowel with accent and since accentuating a V is complicated we are going to change it to A and Á . There is another problem that in each language is accented in a different way and detecting syllables and accents in all languages is going to be very complicated. I will choose Spanish to detect syllables and accents because together with Japanese it is one of the fastest languages and has very standardized the subject, also because I have a simple application to use. Anyway we will leave the possibility for the user to accentuate. You can also use the consonants, the type of vowel or include other modes such as Morse as you have very well devised and developed, Abu. But more on that later. Now an example with the accent:

Proper name "Isabelle". The Silabajs application returns:

{"word": "isabelle", "wordLength":8, "numberSyllable":4, "syllables":[{"startPosition":0, "syllable": "i"},{"startPosition":1, "syllable": "sa"},{"startPosition":3, "syllable": "be"},{"startPosition":5, "silaba": "lle"}], "tonica":3, "letraTildada":-1, "accentuation": "Grave (Llana)", "hiato":[], "diphthong":[], "triphthong":[]}

The simplification would be: ISABELLE A-CA-CÁ-CCA . I don't know how it would be in other languages. Oops! another problem I wouldn't know how to solve consonants or vowels in a row! for example. This for later

Now it's time to start the rhythm search.

**Comment by josepssv** [comment]
Each proper noun, sender or receiver must affect any word Solresol.
The sender and receiver can affect the rhythm at the beginning of the song With the subject "I" we would have **the simplest** and monotonous rhythm 
For the rhythm we will take only vowels and syllables, and so it would be like this 
I = Á Á-Á-Á-Á-Á-Á-Á-Á-Á each Á can have different note times all in the same interval and volume 
Any Solresol word will be altered by the rhythm leaving aside the grammar
For example: 
Syllable: Larefa would be La-re-fa-la-re-la-re-fa-la-re-fa and can be interpreted in different tempos 2 2 2 2 2 2 2 2 2 2 2 or 4 4 4 4 4 4 4 4 4 4 4 or 8 8 8 8 8 8 8 8 8 8 8 8 8 etc. 
If the proper name were for example David = Á-A 
it affects Larefa thus lare-fa-lare-fa-lare-fa =  4 4 2 4 4 2 4 4 2 
if David would be accented A-Á larefa would be la-refa-la-refa-refa-la-refa = 2 4 4 2 4 4 2 4 4
**The accent compresses the notes** 
Another example Camile = A-Á-A => la-refa-la-re-fala-re-fa-lare-fa = 2 4 4 2 2 4 4 2 2 4 4 2 2 
Another proper name Benjamin = Á-A-A => lare-fa-la-refa-refa-la-re-fala-re-fa-re-fa = 4 4 2 2 4 4 2 2 4 4 2 2 4 4 
Once we have the time string that has been formed from the proper name, we form the bar 
For example Á-A-A => 4 4 2 2 4 4 2 2 4 4 2 2 => 4 4 2 | 2 4 4 | 2 2 4 | 4 2 2 
But what is the function to do it?
This is impossible to adjust in all cases. Another option is that instead of a stressed syllable, it slows down the rhythm instead of increasing it:
Benjamin = Á-A-A => 2 4 4 2 4 4 2 4 4 2 4 4 => 2 4 4 | 2 4 4 | 2 4 4 | 2 4 4
Let's see more cases ...

Also to change the pace

I am going to develop this

Done, but it is not always possible to adjust to a 4/4 time signature without leaving gaps.
But the idea of translating accents into rhythms is still pending.

**Comment by josepssv** [comment]
How to use these two elements to make an introduction? 
The introduction can be based on the emitter and the receiver.
But how to transfer letters to notes if Solresol doesn't do it?
Example:
Short duration A : 4 
Long duration Á : 2 
We have emitter="Me"  receiver="Oliver"
Emitter accents=Á  receiver accents= ÁAA ;      ÁÁAA = 2244
Solresol: Laredofa ADCF (blonde) 
2244 ADCF
2A 2D 4C 4F
